lib/compress/zstd_compress.c

```diff
@@ -2568,8 +2572,9 @@ static size_t ZSTD_compress_frameChunk (ZSTD_CCtx* cctx,
      * greater than the optimistic number of blocks we still have remaining.
      * This might be UNset when data is uncompressable and we're streaming.  */
 
-    const int enoughDstCapacityForNoCompressSuperBlocks =
-        (dstCapacity / (blockSize + 7 /* header + checksum */)) > (srcSize / blockSize);
+    const size_t targetCBlockSize = cctx->appliedParams.targetCBlockSize;
+    const int enoughDstCapacityForNoCompressSuperBlocks = !targetCBlockSize ? 0 :
```

Comment:
Is it possible for `targetCBlockSize` to be `== 0` at this point in  the code ?
If not, this should be an `assert()`, rather than a permanent test.

Reply:
Yes, it is possible for `targetCBlockSize == 0`. I had to include this because fuzzer.c crashed so having an assert here would do that same. Maybe we should enforce `targetCBlockSize != 0` before it reaches this function and add an assert here instead?

Reply:
Well, I confused the position in the source code : reading `enoughDstCapacityForNoCompressSuperBlocks`, I assumed this code was in a superblock-dedicated function. But it's not, it's in `ZSTD_compress_frameChunk()`, which is generic, and therefore `targetCBlockSize` can be `0`, meaning "no superblock".

However, this asks another question : why is this test done in `ZSTD_compress_frameChunk()` ?

The way I understand the code, the codec will now disable superblock when `dstCapacity` is not large enough for the pessimistic case. This feels a bit too strong. Especially, I'm worried that "normal" scenarios, such as "average size" data (~100 KB) receiving `dstCapacity = ZSTD_compressBound(srcSize)`, may end up constantly disabling superblock. That would defeat the purpose of superblock.

Reply:
Yeah, that's a good point. It might be a little too sensitive to put it here. I'll move it inside superblock